packages/@aws-cdk/aws-eks/lib/cluster-resource.ts

```diff
@@ -0,0 +1,83 @@
+import cfn = require('@aws-cdk/aws-cloudformation');
+import { PolicyStatement } from '@aws-cdk/aws-iam';
+import iam = require('@aws-cdk/aws-iam');
+import lambda = require('@aws-cdk/aws-lambda');
+import { Construct, Duration, Token } from '@aws-cdk/core';
+import path = require('path');
+import { CfnClusterProps } from './eks.generated';
+import { KubectlLayer } from './kubectl-layer';
+
+/**
+ * A low-level CFN resource Amazon EKS cluster implemented through a custom
+ * resource.
+ *
+ * Implements EKS create/update/delete through a CloudFormation custom resource
+ * in order to allow us to control the IAM role which creates the cluster. This
+ * is required in order to be able to allow CloudFormation to interact with the
+ * cluster via `kubectl` to enable Kubernetes management capabilities like apply
+ * manifest and IAM role/user RBAC mapping.
+ */
+export class ClusterResource extends Construct {
+  /**
+   * The AWS CloudFormation resource type used for this resource.
+   */
+  public static readonly RESOURCE_TYPE = 'Custom::AWSCDK-EKS-Cluster';
+
+  public readonly attrEndpoint: string;
+  public readonly attrArn: string;
+  public readonly attrCertificateAuthorityData: string;
+  public readonly ref: string;
+
+  /**
+   * The IAM role which created the cluster. Initially this is the only IAM role
+   * that gets administrator privilages on the cluster (`system:masters`), and
+   * will be able to issue `kubectl` commands against it.
+   */
+  public readonly creationRole: iam.IRole;
+
+  constructor(scope: Construct, id: string, props: CfnClusterProps) {
+    super(scope, id);
+
+    // each cluster resource will have it's own lambda handler since permissions
+    // are scoped to this cluster and related resources like it's role
+    const handler = new lambda.Function(this, 'ResourceHandler', {
+      code: lambda.Code.asset(path.join(__dirname, 'cluster-resource')),
+      runtime: lambda.Runtime.PYTHON_3_7,
+      handler: 'index.handler',
+      timeout: Duration.minutes(15),
+      memorySize: 512,
+      layers: [ KubectlLayer.getOrCreate(this) ],
+    });
+
+    if (!props.roleArn) {
+      throw new Error(`"roleArn" is required`);
+    }
+
+    // since we don't know the cluster name at this point, we must give this role star resource permissions
```

Comment:
One thing I can do is allocate the cluster name during synth (e.g. based on the uniqueId of the node) instead of during deployment. Then we can scope down only to this specific cluster. Thoughts?